### ISSUE TITLE
poweredup device list hangs

### ISSUE THREAD

**Issue by mennolodder** [sharpbrick/powered-up#193]
As mentioned in the other thread, I think this is a separate problem, so will make it's own ticket. 

The commandline tool hangs on receiving messages, for instance when running `poweredup device list`. It does work when I detach all my motors. 

My configuration is mentioned [here](https://github.com/sharpbrick/powered-up/issues/58#issuecomment-1874470180)

Tried with the latest release, as well as the latest main. 

When I enable debugging I get the following output (a few of the log messages are mine):

```
Scan Started. Please select the Hub (using a number keys or 'q' to terminate):
1: TechnicMediumHub (with address 90:84:2B:63:D2:B5)
1
Selected TechnicMediumHub with key 1
Discover Ports. Receiving Messages ...
info: SharpBrick.PoweredUp.Protocol.LegoWirelessProtocol[0]
      Connected to device, getting protocol information
Discovering ports.
info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Number of Ports announced: 13
info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Port 0-0
..info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-0
.......info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-1
.......info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-2
.......info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-3
.......info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-4
.......info: SharpBrick.PoweredUp.Functions.DiscoverPorts[0]
      Discover Mode 0-0-5
.
```

**Comment by mennolodder** [comment]
It hangs on the message: PortModeInformationForRawMessage, for hub 0, port 0 modeIndex 5.
KnowledgeManager.ApplyStaticProtocolKnowledge(await _protocol.SendMessageReceiveResultAsync<PortModeInformationForRawMessage>(new PortModeInformationRequestMessage(port.PortId, modeIndex, PortModeInformationType.Raw) { HubId = _hubId }), knowledge);

Thats the TechnicLargeLinearMotor on port 0, if I unplug it, it does the same for the other TechnicLargeLinearMotors.


For the WeDo 2 it gives the following error (I added the raw data of the sending and receiving, this is the Name message (PortModeInformationType.Name), both in request and reply.:

      Sending 06:00:22:02:00:00
      Received 11:00:44:02:00:00:4C:50:46:32:2D:4D:4D:4F:54:4F:52

      Exception in LegoWirelessProtocol Decode/Knowledge
      System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
         at SharpBrick.PoweredUp.Protocol.Formatter.PortModeInformationEncoder.Decode(Byte hubId, Span`1& data) in C:\Repos\powered-up\src\SharpBrick.PoweredUp\Protocol\Formatter\PortModeInformationEncoder.cs:line 21
         at SharpBrick.PoweredUp.Protocol.Formatter.PortModeInformationEncoder.SharpBrick.PoweredUp.Protocol.Formatter.IMessageContentEncoder.Decode(Byte hubId, Span`1& data)
         at SharpBrick.PoweredUp.Protocol.Formatter.MessageEncoder.Decode(Span`1& data, ProtocolKnowledge knowledge) in C:\Repos\powered-up\src\SharpBrick.PoweredUp\Protocol\Formatter\MessageEncoder.cs:line 86
         at SharpBrick.PoweredUp.Protocol.LegoWirelessProtocol.<ConnectAsync>b__15_0(Byte[] data) in C:\Repos\powered-up\src\SharpBrick.PoweredUp\Protocol\LegoWirelessProtocol.cs:line 55

**Comment by mennolodder** [comment]
So I dug into the decode error, it seems that some motors don't follow the specs:

![image](https://github.com/sharpbrick/powered-up/assets/7911897/3a892097-4d76-4cb9-841f-534f8c7beb2f)

My Large TechnicLineairMotor  all zero fill these messages, instead of taking variable length (these messages match your dumps):
Example: 11:00:44:00:04:00:4C:4F:41:44:00:00:00:00:00:00:00

For the wedo 2.0 motor the name is 'LPF2-MMOTOR' which fills all characters so doesn't have a 0x00 to terminate on.

I'll see if I can make a fix for it and make a PR.

**Comment by mennolodder** [comment]
I've been able to debug this:

It goes wrong here:
```
dbug: SharpBrick.PoweredUp.Bluetooth.BluetoothKernel[0]
      > 06-00-22-00-05-00
dbug: SharpBrick.PoweredUp.Bluetooth.BluetoothKernel[0]
      < 05-00-05-22-06
info: SharpBrick.PoweredUp.Protocol.LegoWirelessProtocol[0]
      Received 05:00:05:22:06
fail: SharpBrick.PoweredUp.Functions.TraceMessages[0]
      Error - InvalidUse from PortModeInformationRequest
```
The reply translates as: Error - Port Mode Information Request - Port 0 - Invalid use (e.g. parameter error(s)

Which makes sense, because it is requesting mode 5 (so the 6th mode) where earlier the device announced 5 modes (( checked, the bold number is 'Total Mode Count'_:

```
dbug: SharpBrick.PoweredUp.Bluetooth.BluetoothKernel[0]
      > 05-00-21-00-01
dbug: SharpBrick.PoweredUp.Bluetooth.BluetoothKernel[0]
      < 0B-00-43-00-01-0F-**05**-1E-00-1F-00
info: SharpBrick.PoweredUp.Functions.TraceMessages[0]
      Port Information - Port 0/0 Total Modes 5 / Capabilities Output:True, Input:True, LogicalCombinable:True, LogicalSynchronizable:True / InputModes: 1E, OutputModes: 1E
```

I didn't doublecheck this, but this probably is because the TechnicLargeLinearMotor.GetStaticPortInfoMessages announces 6 modes:
0B-00-43-00-01-0F-**06**-1E-00-1F-00

**Comment by mennolodder** [comment]
Update, just because I wondered why this wasn't closed.

This issue was twofold:
1. closed in #194 to support messages that don't seem to follow specs
2. still open #197 to update the technic large motor.
